File: python/problem12.py
import math

def countDivisors(num):
    incr = 1
    if num % 2 != 0: incr = 2
    count = 2
    if num % 2 == 0: count = 4
    for i in range(3, int(math.sqrt(num))+1, incr):
        if num%i == 0: count += 2
    return count

def findHighlyDivisibleTriangularNumber(limit):
    triNum = 1
    incr = 2
    while True:
        triNum += incr
        if countDivisors(triNum) >= limit: return triNum
        incr += 1

if __name__=='__main__':
    print(findHighlyDivisibleTriangularNumber(5))

File: go/problem12.go
package main

import (
	"fmt"
	"math"
)

func main() {
	v, err := FindHighlyDivisibleTriangularNumber(5)
	if err != nil {
		fmt.Println("Error : ", err)
	}
	fmt.Println(v)
}

func divisorCount(num int) int {
	
	incr := 2
	count := 2
	if num%2 == 0 {
		incr = 1
		count = 4
	}

	for i := 3; i <= int(math.Trunc(math.Sqrt(float64(num)))); i += incr {
		if num%i == 0 {
			count += 2
		}
	}

	return count
}

func FindHighlyDivisibleTriangularNumber(limit int) (int, error) {

	triNum := 1
	incr := 2
	for true {
		triNum = triNum + incr
		//fmt.Printf(" %v ", triNum)
		if divisorCount(triNum) >= limit {
			return triNum, nil
		}
		incr++
	}

	return -1, fmt.Errorf("Could not find a triangular number with %v divisors", limit)
}
